### ISSUE TITLE
Less error prone form of INSTANCENAME

### ISSUE THREAD

**Issue by KSchopmeyer** [pywbem/pywbemtools#390]
The INSTANCENAME as WBEM URI is difficult to enter an error prone.  The use of  wildcard helps but is interactive.

One of our goals is to make pywbemcli as easily usable as possible, in particular make the command line input both intuitive and as free of specialized formats, characters, etc. as possible. Through experience with cimcli it became obvious that entering the CIMInstanceName parameter manually in a form acceptable to the WBEM URI standards was messy, partly because of the requirements for the double-quotes which must be escaped from most command line environments. The instance name is messy, involving multiple key properties and having to enter each with escaped double quotes for all non numerics caused users, especially casual users headaches in getting everything correct. This was compounded by the fact that a malformed instance name input on the command line simply generates an error that it was malformed.

There are several ways to make this easier including:

1. The wildcard which we already have which uses the classname with a wildcard "?* for the keybindings component and pywbemcli presents a list of possible known instance names.

2. Eliminate the need for the quotes as much as possible. We can make a form that is easier to input by eliminating the need for the quote marks in most cases.  This would use a parameter similar to the --property in instance create but it would be --key where each use would be a single key.  pywbemcli would assemble the CIMInstanceName from the individual keys including addin sayg quotes, etc.

3. Lead the user through the process step by step where pywbemcli would determine the key properties and present each key property with its type and ask the user to input a string representing the value

**Comment by andy-maier** [comment]
I don't expect that people will type in WBEM URI strings, other than copying them from somewhere.

**Comment by KSchopmeyer** [comment]
This proposal is that we use an option to build the keys.   KARL FINISH THIS

TODO exlain association keys.

**Comment by KSchopmeyer** [comment]
There is another mechanism we could use, similar to the wildcard but much more interactive, an instancename creator that would ask the user for the value of each key. For example: If we the user must build up a key with multiple keys we could identify the requirement with a wildcard tag on the keybindings part of the instance name and then get the class from the server and ask for the value of each key.

instance get CIM_Person.*     # make * the wildcard to make detailed request for sake of this example:

```

  Enter value or just hit Enter to ignore this key. Hit Ctrl-C to abort:
   CreationCllassName(String) >  <user enters value>
   Name(string)> <User enter value>
  
   instancename is CIM.Person.CreationClassName="blah",Name="me"
   ... proceeds to get the instance from the server

```

**Comment by andy-maier** [comment]
Fixed in PR #455.
